### PR TITLE
Add test scenarios for gas fee

### LINKED ISSUE #18947: Gas Fees: legacy and eip1559 supported estimates

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

Add new test scenarios for gas fee.

## **Related issues**

Closes: #18947 

## **Manual testing steps**

Not necessary.

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

### CODE REVIEW

File: test/scenarios/11. gas fee/EIP-1559 gas.csv [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+8,"On notification screen, click ""Site suggested"" button.",,"Edit gas fee screen is shown. Three gas options are shown: low, market, aggressive. Site suggested gas fee is shown. ""Advanced"" button is displayed. ",
+9,"On edit gas fee screen, click ""Advanced"" button.",,Advanced gas fee edit screen is shown. Editable max base fee and priority fee input fields are displayed.,
+10,"In max base fee and priority fee input fields, enter new values, and click ""Save"".",e.g. 25 for max base fee and 2 for priority fee,"On notification screen, the displayed gas fee is different from previous one.","Take a note for the new gas fee. "
+11,"On notification screen , click ""Confirm"" to proceed the transaction.",,The transaction appears in the activity list. The recipient's hexadecimal address is shown in the activity list item. The amount is shown in the activity list item.,
+12,Open the activity list item.,,"The transaction status, recipient's address, nonce, amount are shown in the item details. Values of displayed total gas fee, priority fee, and max fee per gas are the new values what we changed in previous step.",
```
Comment (DDDDDanica) [inline test/scenarios/11. gas fee/EIP-1559 gas.csv:17]:
```suggestion
12,Open the activity list item.,,"The transaction status, recipient's address, nonce, amount are shown in the item details. Values of displayed total gas fee, priority fee, and max fee per gas are the new values that we changed in previous step.",
```

File: test/scenarios/11. gas fee/legacy gas.csv [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+6,Connect with the current account.,,,
+7,"In test dapp, click ""send legacy transaction"".",,MetaMask notification screen is shown. Estimated gas fee is displayed. Edit button is displayed.,Take a note for the estimated gas fee.
+8,"On notification screen, click ""Edit"".",,"Edit screen is shown. Gas fee information message is shown. ""Edit suggested gas fee"" button is displayed. ",
+9,"On edit screen, click ""Edit suggested gas fee"".",,Editable gas limit and gas price input fields are displayed.,
+10,"In gas price input field, enter a different value, and click ""Save"".",e.g. 50,"On notification screen, the displayed gas fee is different from previous one.","Take a note for the current gas fee. "
+11,"On notification screen , click ""Confirm"" to proceed the transaction.",,The transaction appears in the activity list. The recipient's hexadecimal address is shown in the activity list item. The amount is shown in the activity list item.,
+12,Open the activity list item.,,"The transaction status, recipient's address, nonce, amount are shown in the item details. Value of displayed gas price and total gas fee are the new values what we changed in previous step.",
```
Comment (DDDDDanica) [inline test/scenarios/11. gas fee/legacy gas.csv:17]:
```suggestion
12,Open the activity list item.,,"The transaction status, recipient's address, nonce, amount are shown in the item details. Value of displayed gas price and total gas fee are the new values that we changed in previous step.",
```

File: test/scenarios/11. gas fee/EIP-1559 gas.csv (hunk 1 above)
Comment (chloeYue) [reply test/scenarios/11. gas fee/EIP-1559 gas.csv:17]: Done. Thanks !

**Review by DDDDDanica — DISMISSED** [review]

File: test/scenarios/11. gas fee/EIP-1559 gas.csv [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+5,Proceed to Connect with MetaMask.,,,
+6,Connect with the current account.,,,
+7,"In test dapp, click ""send EIP 1559 transaction"".",,"MetaMask notification screen is shown. Current gas fee is displayed. ""Site suggested"" button is displayed.",Take a note for the current gas fee.
+8,"On notification screen, click ""Site suggested"" button.",,"Edit gas fee screen is shown. Three gas options are shown: low, market, aggressive. Site suggested gas fee is shown. ""Advanced"" button is displayed. ",
+9,"On edit gas fee screen, click ""Advanced"" button.",,Advanced gas fee edit screen is shown. Editable max base fee and priority fee input fields are displayed.,
+10,"In max base fee and priority fee input fields, enter new values, and click ""Save"".",e.g. 25 for max base fee and 2 for priority fee,"On notification screen, the displayed gas fee is different from previous one.","Take a note for the new gas fee. "
+11,"On notification screen , click ""Confirm"" to proceed the transaction.",,The transaction appears in the activity list. The recipient's hexadecimal address is shown in the activity list item. The amount is shown in the activity list item.,
```
Comment (PeterYinusa) [inline test/scenarios/11. gas fee/EIP-1559 gas.csv:16]: > The recipient's hexadecimal address is shown in the activity list item.
I hexadecimal address is not shown here

File: test/scenarios/11. gas fee/legacy gas.csv [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+The wallet address is shown on the overview.
+The selected network is Sepolia.",
+4,Open the test dapp in another tab.,https://metamask.github.io/test-dapp/,,
+5,Proceed to Connect with MetaMask.,,,
+6,Connect with the current account.,,,
+7,"In test dapp, click ""send legacy transaction"".",,MetaMask notification screen is shown. Estimated gas fee is displayed. Edit button is displayed.,Take a note for the estimated gas fee.
+8,"On notification screen, click ""Edit"".",,"Edit screen is shown. Gas fee information message is shown. ""Edit suggested gas fee"" button is displayed. ",
+9,"On edit screen, click ""Edit suggested gas fee"".",,Editable gas limit and gas price input fields are displayed.,
+10,"In gas price input field, enter a different value, and click ""Save"".",e.g. 50,"On notification screen, the displayed gas fee is different from previous one.","Take a note for the current gas fee. "
+11,"On notification screen , click ""Confirm"" to proceed the transaction.",,The transaction appears in the activity list. The recipient's hexadecimal address is shown in the activity list item. The amount is shown in the activity list item.,
```
Comment (PeterYinusa) [inline test/scenarios/11. gas fee/legacy gas.csv:16]: > The recipient's hexadecimal address is shown in the activity list item.

The hexadecimal address is not shown here

File: test/scenarios/11. gas fee/EIP-1559 gas.csv (hunk 3 above)
Comment (chloeYue) [reply test/scenarios/11. gas fee/EIP-1559 gas.csv:16]: Nice catch, corrected. Thanks!

File: test/scenarios/11. gas fee/legacy gas.csv (hunk 4 above)
Comment (chloeYue) [reply test/scenarios/11. gas fee/legacy gas.csv:16]: Nice catch, corrected. Thanks!

**Review by PeterYinusa — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**